### DISCUSSION TITLE
I didn't understand receive and fallback functions

### DISCUSSION THREAD

**Discussion by fadai88** [Cyfrin/foundry-full-course-cu#1266] [Q&A]
I didn't understand receive and fallback functions. If we have send or transfer functions, why do we need receive or fallback?
Could anyone please explain it?

**Comment by xyz899** [comment]
Hi,

Receive and fallback functions are very simple, let's say you have a case of scenario where someone send Ether or other crypto to a smart contract that does not have a fallback or receive function or even a payable function, and receives Ether, the transaction will fail and revert. This is because the Ethereum Virtual Machine (EVM) requires some kind of function to handle incoming transactions. 

```
contract example {
   fallback() external payable {}

   receive() external payable {}
   
     /*
    Which function is called, fallback() or receive()?

           send Ether
               |
         msg.data is empty?
              / \
            yes  no
            /     \
receive() exists?  fallback()
         /   \
        yes   no
        /      \
    receive()   fallback()
    */
}
```


So `receive()` acts as a receiver of valued data and store the amount in the contract and `fallback()` triggers when a receive function is not included, so one protects the other

**Reply by fadai88** [reply]
Thank you!

**Reply by xyz899** [reply]
no problem, don't forget to mark the discussions as solved by answer

**Reply by xyz899** [reply]
you didn't select the answer, select the answer when you have a correct one so people know there is one
